src/base.jl

```diff
@@ -37,15 +37,86 @@ Base.lastindex(a::AbstractMPSorMPO) = lastindex(a.tensors)
 Base.length(a::AbstractMPSorMPO) = length(a.tensors)
 Base.size(a::AbstractMPSorMPO) = (length(a.tensors), )
 
+LinearAlgebra.rank(ψ::MPS) = Tuple(size(A, 2) for A ∈ ψ)
 
-function MPS(vec::Vector{Vector{T}}) where  {T <: Number}
-    L = length(vec)
+MPS(A::AbstractArray) = MPS(A, :right)
+MPS(A::AbstractArray, s::Symbol, args...) = MPS(A, Val(s), typemax(Int), args...)
+MPS(A::AbstractArray, s::Symbol, Dcut::Int, args...) = MPS(A, Val(s), Dcut, args...)
+MPS(A::AbstractArray, ::Val{:right}, Dcut::Int, args...) = _left_sweep_SVD(A, Dcut, args...)
+MPS(A::AbstractArray, ::Val{:left}, Dcut::Int, args...) = _right_sweep_SVD(A, Dcut, args...)
```

Comment:
I'm not sure if this gonna work that easily. In our context, everything will be instance-dependent and one good config to "rule them all" will not do. I think we should leave the back door into psvd, just in case, or at least for now.